File: c/armstrong-numbers/armstrong_numbers.c
#include "armstrong_numbers.h"

bool is_armstrong_number(int num) {
    int num_digits = 0;
    int temp = num;
    while (temp > 0) {
        temp /= 10;
        ++num_digits;
    }

    int ans = 0;
    temp = num;
    while (temp > 0) {
        int digit = temp % 10;
        int pow = digit;
        for (int i = 1; i < num_digits; i++) {
            pow *= digit;
        }
        ans += pow;
        temp /= 10;
    }

    return ans == num;
}

File: zig/armstrong-numbers/armstrong_numbers.zig
const std = @import("std");

pub fn isArmstrongNumber(num: u128) bool {
    var buf: [64]u8 = undefined;
    const strNum = std.fmt.bufPrint(&buf, "{d}", .{num}) catch unreachable;
    const numDigits: usize = strNum.len;

    var n: u128 = num;
    var sum: u128 = 0;
    while (n > 0) {
        const digit: u128 = n % 10;
        sum += std.math.pow(u128, digit, numDigits);
        n /= 10;
    }
    return sum == num;
}
